gcc/rust/expand/rust-macro-builtins-format-args.cc

```diff
@@ -162,7 +163,8 @@ MacroBuiltin::format_args_handler (location_t invoc_locus,
   if (append_newline)
     fmt_str += '\n';
 
-  auto pieces = Fmt::Pieces::collect (fmt_str, append_newline);
+  auto pieces = Fmt::Pieces::collect (fmt_str, append_newline,
+				      Fmt::ffi::ParseMode::Format);
```

Comment:
another small nitpick - on the C++ side of things I'd like to avoid using the `ffi` namespace, as it's meant more for FFI related stuff. so ideally we'd have two functions, something like `collect_asm_pieces` and `collect_fmt_pieces` (or other names). but again, this is a suggestion and not a requirement :)